server-setup/audit-event-functions.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  type AuditEvent,
+  type AuditEventAgent,
+  type AuditEventSource,
+  type AuditEventAction,
+  type Coding,
+  type FhirClient,
+  type Retrieved,
+  type AuditEventEntity,
+  type AuditEventOutcome,
+  FetchFhirClient,
+} from "@bonfhir/core/r4b";
+
+// Audit Event Success Token
+const SUCCESS = "0";
+
+// Terminology Source
+export type TerminologySource = {
+  source: string;
+  system: string;
+  version: string;
+};
+
+export async function getApplicationImportAuditEvent(
+  baseUrl: string
+): Promise<Retrieved<AuditEvent> | undefined> {
+  const client: FhirClient = new FetchFhirClient({
+    baseUrl,
+  });
+  // TODO: this should return all matching audit event and we need to parse the entity detail to match
+  try {
+    const bundle = await client.search("AuditEvent", (search) => {
+      const importOntology: Coding = {
+        system: "http://dicom.nema.org/resources/ontology/DCM",
+        code: "110107",
+      };
+      return search.outcome("0").type(importOntology);
+    });
+    return bundle.bundle.entry?.[0].resource;
+  } catch (error) {
+    console.error("Error fetching audit event", error);
+    return undefined;
+  }
```

Comment:
that function could cache the bundle in a module variable and return it on later calls instead of fetching

Reply:
✅